WalletWasabi/Crypto/ZeroKnowledge/ProofSystem.cs

```diff
@@ -40,5 +41,35 @@ public static Statement IssuerParameters(CoordinatorParameters iparams, MAC mac,
 				{ Generators.GV - iparams.I, O,             O,              Generators.Gx0, Generators.Gx1, Generators.Ga },
 				{ iparams.Cw,                Generators.Gw, Generators.Gwp, O,              O,              O },
 			});
+
+		public static Knowledge SerialNumber(GroupElement Ca, GroupElement S, Scalar z, Scalar a, Scalar r)
+			=> new Knowledge(SerialNumber(Ca, S), new ScalarVector(z, a, r));
+
+		public static Statement SerialNumber(GroupElement Ca, GroupElement S)
+			=> new Statement(new GroupElement[,]
+			{
+				// public                    Witness terms:
+				// point      z              a               r
+				{ Ca,         Generators.Ga, Generators.Gg,  Generators.Gh },
+				{ S,          O,             O,              Generators.Gs }
+			});
+
+		public static (GroupElement Ca, GroupElement S) SerialNumberPublicPoints(Scalar z, Scalar a, Scalar r)
```

Comment:
maybe `ComputeSerialNumberPublicPoints` for consistency with `ComputeZ`? in which case it's perhaps a bit more logical to `ComputeSerialNumber` and just return `S`? The real calling code won't need another copy of `Ca` as that would already be provided by the randomization so i think this return type reflects the artificial needs of the tests